pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -608,6 +610,29 @@ func (cb *ClusterBuilder) setUpstreamProtocol(cluster *clusterWrapper, port *mod
 		return
 	}
 
+	// Preserve HTTP/1.x traffic header case
+	if port.Protocol.IsHTTP() && cb.req.Push.ProxyConfigs.EffectiveProxyConfig(cb.proxyMetadata, cb.req.Push.Mesh).GetProxyHeaders().GetPreserveHttp1HeaderCase().GetValue() {
```

Comment:
> IsHTTP()

I think we need it for auto-detected as well? IsUnsupported()

Reply:
Fixed!